docs/modules/ROOT/pages/references/parameters.adoc

```diff
@@ -173,6 +164,134 @@ This gives you the full control over the resulting MachineSet.
 Values given here will be merged with precedence with the defaults configured in <<defaultSpec>>.
 The values can be everything that's accepted in the `spec` field of a `machinesets.machine.openshift.io` object.
 
+== `machineConfigPools`
+
+[horizontal]
+type:: dictionary
+
+A dictionary of machine config pools to create on the cluster.
+The resulting `MachineConfigPool` object will have the key prefixed with `x-` as the name and the value is merged into the resource.
+
+Apart from the machine config pool, this dictionary can manage related resources through the following fields, that won't be added to the `MachineConfigPool` object.
+
+* The **kubelet** field can contain the spec of a `KubeletConfig` resource.
+The machine config pool key will be used as the name and the `machineConfigPoolSelector` will be set automatically.
+* The **containerRuntime** field can contain the spec of a `ContainerRuntimeConfig` resource.
+The machine config pool key will be used as the name and the `machineConfigPoolSelector` will be set automatically.
```

Comment:
I don't like having multiple ways to specify containerRuntimeConfigs and kubeletConfigs, but I guess it's fine for now.

If we keep the implementation as-is for now, I'd like to have a follow-up issue to mark the top-level `kubeletConfigs` and `containerRuntimeConfigs` fields as deprecated so that we can remove them at some point.

Reply:
We probably shouldn't just remove the old parameters without a deprecation period, since I assume that there's a number of configurations which actually make use of them. I'd much prefer spending some extra time in a follow-up issue / PR to make the migration as painless as possible on the config side (regardless of whether a maintenance window is required for this PR).

Reply:
Opened https://github.com/appuio/component-openshift4-nodes/issues/51 as a follow up